Remove duplicate c++std library

ShootingKing-AM opened the pull request:
- Fixes https://github.com/alvr-org/alvr-cardboard/issues/3 
- Static c++stdlib is included already here, no need for linking shared c++stdlib again - https://github.com/alvr-org/ALVR/blob/528fe67df4a8f7ff980da0a8ebe728a627069d29/alvr/client_core/build.rs#L25-L34

zarik5 commented:
The curious thing is that I added that line not too long ago to fix a crash on the smartphone/Monado version. I'm not sure how to reconcile this.

ShootingKing-AM commented:
> The curious thing is that I added that line not too long ago to fix a crash on the smartphone/Monado version. I think we should control this using a cargo feature flag, and expose that from xtask too.

Can you link to the issue ? we can control it that way, but it really has c++static stdlib already in it, and it shouldn't ask for a new lib

ShootingKing-AM commented:
> The curious thing is that I added that line not too long ago to fix a crash on the smartphone/Monado version. I think we should control this using a cargo feature flag, and expose that from xtask too.

Added shared stdlibc++ linking as a feature as suggested :)

zarik5 reviewed (approved):
Since this PR does not integrate with xtask to conditionally add the flag, it essentially disables it completely. I tested this on my Quest Pro and it works still. I think we can merge this and fix Monado compatibility another time.
